deploy/modo-server/server.py

```diff
@@ -41,6 +44,24 @@ def gather_metadata():
         archive = zarr.open(
             store=store,
         )
-        meta = MODO(path=f"{S3_LOCAL_URL}/{modo}", archive=archive).metadata
+        meta.update(
+            MODO(path=f"{S3_LOCAL_URL}/{modo}", archive=archive).metadata
+        )
 
     return meta
+
+
+@app.get("/get")
+def get_s3_path(query: str, exact_match: bool = False):
+    """Receive the S3 path of all modos matching the query"""
+    modos = minio.ls(BUCKET)
+    if not exact_match:
+        res = [modo for modo in modos if query in modo]
+    else:
+        # NOTE: Is there a better/more roboust way than regexpr?
+        res = [
+            modo
+            for modo in modos
+            if re.search("/" + query + r"$", modo) is not None
+        ]
```

Comment:
**suggestion:** just found out about [`difflib.SequenceMatcher()`](https://docs.python.org/3/library/difflib.html#sequencematcher-objects), which is in python's standard library:

```python
>>> import difflib
>>> difflib.SequenceMatcher(None, 'example', 'example').ratio()
1.0
>>> difflib.SequenceMatcher(None, 'example', 'exemple').ratio()
0.8571428571428571
>>> difflib.SequenceMatcher(None, 'example', 'many examples').ratio()
0.7
>>> difflib.SequenceMatcher(None, 'lion', 'zebra').ratio()
0.0
```
We could maybe just apply a threshold on it?
There's also a `.real_quick_ratio()` which uses a heuristic to go much faster. We could also use this one?

Reply:
```python
>>> %timeit difflib.SequenceMatcher(None, 'the lazy dog jumps over the lazy dog', 'the quick brown fox jumps over the lazy dog').ratio()
82.7 µs ± 5.29 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

>>> %timeit difflib.SequenceMatcher(None, 'the lazy dog jumps over the lazy dog', 'the quick brown fox jumps over the lazy dog').real_quick_ratio()
9.83 µs ± 603 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```